templates/azure-cluster-identity/azure-cluster-identity.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
   labels:
     clusterctl.cluster.x-k8s.io/move-hierarchy: "true"
 spec:
-  type: WorkloadIdentity
+  type: UserAssignedMSI
```

Comment:
I rather have this defaulted to `WorkloadIdentity` and set this via env variable for local testing..

Reply:
Rebased and resolved.